### ISSUE TITLE
Execution does not stop on breakpoints.

### ISSUE THREAD

**Issue by AxelBohm** [jupyterlab/jupyterlab#11462]
## Description
Jupyter lab just runs through the code as if the breakpoints weren't there.

Here's a screenshot of what happens after I execute the two example cells from the tutorial. In particular, it just gives me 
![debugger](https://user-images.githubusercontent.com/28197445/141484572-f7f477cc-f581-4016-9569-d483123fb231.png)

## Reproduce

`conda create -n new_env -c conda-forge python=3.8 jupyterlab=3 ipykernel xeus-python`

## Expected behavior

I expect code execution to stop at the breakpoint so I can inspect the value of a variable.

## Context

- Operating System and version: Linux arch 5.14.14
- Browser and version: Firefox 93

xeus                             2.2.0                hab3612f_0    conda-forge
xeus-python                 0.13.5           py38hcf90354_0    conda-forge
xeus-python-shell         0.1.7              pyhd8ed1ab_0    conda-forge
jupyterlab                     3.2.3              pyhd8ed1ab_0    conda-forge
jupyterlab_server         2.8.2              pyhd8ed1ab_0    conda-forge
notebook                      6.4.5              pyha770c72_0    conda-forge

## Edit:
I tired the same as above but with python 3.9 and everything worked as expected.

**Comment by fcollonval** [comment]
@AxelBohm are you still seeing the issue?

I successfully debug your example with the following kernel environment:

```
_libgcc_mutex                         0.1  conda_forge         conda-forge/linux-64
_openmp_mutex                         4.5  1_gnu               conda-forge/linux-64
backcall                            0.2.0  pyh9f0ad1d_0        conda-forge/noarch  
backports                             1.0  py_2                conda-forge/noarch  
backports.functools_lru_cache       1.6.4  pyhd8ed1ab_0        conda-forge/noarch  
ca-certificates                2021.10.26  h06a4308_2          pkgs/main/linux-64  
debugpy                             1.5.1  py38h709712a_0      conda-forge/linux-64
decorator                           5.1.0  pyhd8ed1ab_0        conda-forge/noarch  
ipython                            7.30.1  py38h578d9bd_0      conda-forge/linux-64
jedi                               0.18.1  py38h578d9bd_0      conda-forge/linux-64
ld_impl_linux-64                   2.36.1  hea4e1c9_2          conda-forge/linux-64
libffi                              3.4.2  h7f98852_5          conda-forge/linux-64
libgcc-ng                          11.2.0  h1d223b6_11         conda-forge/linux-64
libgomp                            11.2.0  h1d223b6_11         conda-forge/linux-64
libnsl                              2.0.0  h7f98852_0          conda-forge/linux-64
libsodium                          1.0.18  h516909a_1          conda-forge/linux-64
libstdcxx-ng                       11.2.0  he4da1e4_11         conda-forge/linux-64
libuuid                            2.32.1  h14c3975_1000       conda-forge/linux-64
libzlib                            1.2.11  h36c2ea0_1013       conda-forge/linux-64
matplotlib-inline                   0.1.3  pyhd8ed1ab_0        conda-forge/noarch  
ncurses                               6.2  h58526e2_4          conda-forge/linux-64
openssl                            1.1.1l  h7f98852_0          conda-forge/linux-64
parso                               0.8.3  pyhd8ed1ab_0        conda-forge/noarch  
pexpect                             4.8.0  py38h32f6830_1      conda-forge/linux-64
pickleshare                         0.7.5  py38h32f6830_1002   conda-forge/linux-64
pip                                21.3.1  pyhd8ed1ab_0        conda-forge/noarch  
prompt-toolkit                     3.0.24  pyha770c72_0        conda-forge/noarch  
ptyprocess                          0.7.0  pyhd3deb0d_0        conda-forge/noarch  
pygments                           2.10.0  pyhd8ed1ab_0        conda-forge/noarch  
python                             3.8.12  hb7a2778_2_cpython  conda-forge/linux-64
python_abi                            3.8  2_cp38              conda-forge/linux-64
readline                              8.1  h46c0cb4_0          conda-forge/linux-64
setuptools                         60.1.1  py38h578d9bd_0      conda-forge/linux-64
sqlite                             3.37.0  h9cd32fc_0          conda-forge/linux-64
tk                                 8.6.11  h27826a3_1          conda-forge/linux-64
traitlets                           5.1.1  pyhd8ed1ab_0        conda-forge/noarch  
wcwidth                             0.2.5  pyh9f0ad1d_2        conda-forge/noarch  
wheel                              0.37.1  pyhd8ed1ab_0        conda-forge/noarch  
xeus                                2.2.0  hab3612f_0          conda-forge/linux-64
xeus-python                        0.13.5  py38hcf90354_1      conda-forge/linux-64
xeus-python-shell                   0.1.7  pyhd8ed1ab_0        conda-forge/noarch  
xz                                  5.2.5  h516909a_1          conda-forge/linux-64
zeromq                              4.3.4  h9c3ff4c_1          conda-forge/linux-64
zlib                               1.2.11  h36c2ea0_1013       conda-forge/linux-64
```

Using JupyterLab 3.2.5 on Debian 11

![Peek 2021-12-30 14-29](https://user-images.githubusercontent.com/8435071/147756386-7f11896f-1c8c-4b22-a592-beb52971e1c3.gif)

**Comment by chmendoza** [comment]
I had to install ipdb for this to work. Funny that the [docs](https://jupyterlab.readthedocs.io/en/stable/user/debugger.html) don't mention that as a requirement.

**Comment by AxelBohm** [comment]
Ok, it is now one year later and in the meantime I haven't used the debugger in jupyterlab. 

I created the same test environment again : `conda create -n new_env -c conda-forge python=3.8 jupyterlab=3 ipykernel xeus-python` but with a newer version of jupyterlab (3.5.1) and  of the kernels and now now longer observe the issue (even in python 3.8).
Not quite sure which one of the newer versions fixed this.
Btw, I did not have to install `ipdb` for this to work.

Thanks everybody for the great work!
